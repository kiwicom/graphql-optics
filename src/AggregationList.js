import React from 'react'

export default ({ aggregations }) => (
  <ul>
    {aggregations.map(aggregation => (
      <li key={aggregation.key}>
        {aggregation.key} - {aggregation.doc_count}
      </li>
    ))}
  </ul>
)
