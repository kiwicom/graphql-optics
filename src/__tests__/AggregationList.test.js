import React from 'react';
import AggregationList from '../AggregationList';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const data = [ 
    {
      "key" : "allLocations",
      "doc_count" : 6
    },
    {
      "key" : "allFlights",
      "doc_count" : 3
    },
    {
      "key" : "allWhatever",
      "doc_count" : 2
    }
  ];
  const tree = renderer
    .create(<AggregationList aggregations={data}>Facebook</AggregationList>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
