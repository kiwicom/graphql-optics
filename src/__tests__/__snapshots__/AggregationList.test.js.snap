// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <li>
    allLocations
     - 
    6
  </li>
  <li>
    allFlights
     - 
    3
  </li>
  <li>
    allWhatever
     - 
    2
  </li>
</ul>
`;
